PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,585 @@
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
+    
+
+    def makeButtons(self):
+        vcmd = (self.window.register(validate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        findButton = self.addOperation(
+            "Find", lambda: self.clickFind(), numArguments=1,
+            validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1,
+            validationCmd=vcmd)
+        self.addAnimationButtons()
+        deleteButton = self.addOperation(
+            "Delete", lambda: self.clickDelete(), numArguments=0,
+            validationCmd=vcmd, maxRows = 2)
+        newLinkedListButton = self.addOperation(
+            "New", lambda: self.clickNewLinkedList(), 
+            numArguments = 0, validationCmd =vcmd, maxRows = 2)
+        getFirstButton = self.addOperation(
+            "Get First", lambda: self.clickGetFirst(), numArguments = 0,
+            validationCmd=vcmd, maxRows = 2)
+    
+        return [findButton, insertButton, deleteButton, newLinkedListButton, getFirstButton]        
+
+    
+    # validate text entry
+    def validateArgument(self):
+        text = self.getArgument()
+        if text:
+            if text.isdigit():
+                val = int(text)
+                if val< 100000000: return val
+            elif len(text)<9: return text
+        else:
+            return -1
+        
+    def startAnimations(self):
+        self.enableButtons(enable=False)
+        super().startAnimations()    
+        
+    def stopAnimations(self):
+        super().stopAnimations()
+        self.enableButtons(enable=True)
+        self.argumentChanged()           
+            
+#allow letters or numbers to be typed in                  
+def validate(action, index, value_if_allowed,
+            prior_value, text, validation_type, trigger_type, widget_name):
+    if not(value_if_allowed.isdigit()):
+        for i in value_if_allowed:
+            if not 65<=ord(i)<=122 and i not in "0123456789": return False
+    return True
+   
+if __name__ == '__main__':
+    ll = LinkedList()
+    ll.runVisualization()
+
+'''
+Useful Links:
+http://effbot.org/zone/tkinter-complex-canvas.htm
+https://mail.python.org/pipermail/python-list/2000-December/022013.html
+'''
+
+# Just confirming this is working - Etti
+
+#fix arrows..?? long arrow when on last one in delete, and for the actual descending node--**cool effects**
+   #maybe delete each arrow as move node?
+   #or assign it to the previous one's arrow?
+###FIXED: BHH fix delete location rearrangement
+#fix animations upon original insert/setup
+#fix the circles on the nodes
+#fix animation buttons upon the original insert
+     #problem exists bec call insertElem-- which has cleanUp method..
```

Comment:
These notes should be removed.  Since we're using GItHub Issues and PR's to track things to do and what's been done, these just become out of date unless updated with every commit.